### ISSUE TITLE
velero v1.8 taking snapshot twice when CSI plugin is enabled.

### ISSUE THREAD

**Issue by reasonerjt** [vmware-tanzu/velero#4758]
**What steps did you take and what happened:**
1. Install velero v1.8 on GKE:
```
./velero install     --image velero/velero:v1.8.0     --features=EnableCSI     --provider gcp     --plugins velero/velero-plugin-for-csi:v0.2.0,velero/velero-plugin-for-gcp:v1.4.0     --bucket jt-velero     --secret-file ./credentials-velero-gcp
```
2. Backup some workload in a namespace that has PV provisioned by CSI driver (`pd.csi.storage.gke.io`):
```
./velero backup create kibishii-csi-bak --include-namespaces kibishii
``` 
3. By running `velero backup describe` and checking the content on GKE's console, it seems both the volumesnapshotter and the CSI plugin take the snapshot for each of the same PVs.

**What did you expect to happen:**
The CSI plugin should take the snapshot of the PVs and the volmesnapshotter in the GCP plugin will not be triggered for the snapshot.

**The following information will help us better understand what's going on**:
This may be related to the change introduced in v1.8 where we enable the volumesnapshotter of AWS/Azure/GCP to take snapshot of the CSI provisioned PVs by calling cloud providers' API.

_If you are using velero v1.7.0+:_  
Please use `velero debug  --backup <backupname> --restore <restorename>` to generate the support bundle, and attach to this issue, more options please refer to `velero debug --help` 
[debug-csi-snapshot-twice.tar.gz](https://github.com/vmware-tanzu/velero/files/8320111/debug-csi-snapshot-twice.tar.gz)

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Velero version (use `velero version`): v1.8
- Velero features (use `velero client config get features`): 
- Kubernetes version (use `kubectl version`): v1.22
- Kubernetes installer & version:
- Cloud provider or hardware configuration:GKE
- OS (e.g. from `/etc/os-release`):

**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "I would like to see this bug fixed as soon as possible"
- :-1: for "There are more important bugs to focus on right now"

**Comment by aslafy-z** [comment]
I'm hitting the same issue with the AzureDisk plugin along with the CSI plugin.
I'd like to configure the CSI plugin to exclude AzureDisk volumes, is there a way this can be done?

**Comment by blackpiglet** [comment]
By far, there is no way to do that. 
My thought on this issue is when CSI is enabled, skip take snapshot in all cloud provider plugins.

For current situation, my suggestion is not enable both CSI and Azure plugins together, if that is possible for your case.

**Comment by aslafy-z** [comment]
@blackpiglet isn't the provider plugin used to send the backup (meta)data to the Object Storage? I tried to disable the provider plugin but velero is struggling to export the backup content.

**Comment by blackpiglet** [comment]
You are right. Azure plugin is also used as object storage plugin.
I mean how about only use Azure plugin to do the backup and restore for now.
Currently, Azure plugin also support CSI operation, and CSI plugin is still in beta. 
CSI plugin is planning to go GA in v1.9. After that, it will be in more sound state for usage.

**Comment by aslafy-z** [comment]
In my case I have another CSI the same cluster which has no Velero dedicated plugin and implements the CSI VolumeSnapshots capabilities. I have to keep the CSI plugin to work with that one.

**Comment by blackpiglet** [comment]
@aslafy-z 
I think there is not solution for your case by now. After this issue is fixed, it's supposed to be included in v1.9, there should be no duplicated snapshot anymore.

**Comment by fredleger** [comment]
Hello,

hiting this also. i was thincking it could be handy to have a flag on schedule to choose wich type of snapshot you want to use CSI, cloud provider or restic.

My use case: i do want to leverage on CSI snaps to have a local copy but also want an exported copy of volume via restic in case the cloud provider have a major outage (the restic repository is on another cloud provider).

This could be a separate issue if needed. Any thoughts on this ?
